### PR TITLE
[Api] Fixed checkout routing

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: tests/Responses/Expected/checkout/payment_validation_failed.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
                                     "This value is not valid."
                                 ],
                                 "children": [
+                                    {},
```
Comment (lchrusciel) [inline tests/Responses/Expected/checkout/payment_validation_failed.json]: o.O

Comment (pamil) [inline tests/Responses/Expected/checkout/payment_validation_failed.json]: An empty child O.o

File: tests/Responses/Expected/checkout/payment_validation_failed.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
...
                                     {}
```
Comment (pamil) [inline tests/Responses/Expected/checkout/payment_validation_failed.json]: And its brother.

File: tests/Responses/Expected/checkout/payment_validation_failed.json (hunk 1 above)
Comment (lchrusciel) [inline tests/Responses/Expected/checkout/payment_validation_failed.json]: Two of them

File: tests/Responses/Expected/checkout/shipping_validation_failed.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
                                     "This value is not valid."
                                 ],
                                 "children": [
+                                    {},
```
Comment (bretto36) [inline tests/Responses/Expected/checkout/shipping_validation_failed.json]: Empty child here too

**Comment by michalmarcinkowski** [comment]
Please rebase with current master since shipping methods are by `code` not `id` now.

**Comment by michalmarcinkowski** [comment]
Thank you Janek! 👍

**Pull request merged**